From 019a2eb481c34ffd592bb61fec37a7cebd7dc304 Mon Sep 17 00:00:00 2001
From: Radu Voicilas <radu.voicilas@users.noreply.github.com>
Date: Mon, 5 Mar 2018 18:40:43 +0200
Subject: [PATCH] Fix clang int-to-void-pointer-cast warning

warning: cast to 'void *' from smaller integer type 'int' [-Wint-to-void-pointer-cast]
---
 libinotifytools/src/inotifytools.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git libinotifytools/src/inotifytools.c_orig_002 libinotifytools/src/inotifytools.c
index 9fe09bd..fc80d8d 100644
--- libinotifytools/src/inotifytools.c_orig_002
+++ libinotifytools/src/inotifytools.c
@@ -19,6 +19,7 @@
 #include <string.h>
 #include <strings.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <errno.h>
 #include <sys/select.h>
@@ -2086,7 +2087,7 @@ int event_compare(const void *p1, const void *p2, const void *config)
 
 struct rbtree *inotifytools_wd_sorted_by_event(int sort_event)
 {
-	struct rbtree *ret = rbinit(event_compare, (void*)sort_event);
+	struct rbtree *ret = rbinit(event_compare, (void*)(uintptr_t)sort_event);
 	RBLIST *all = rbopenlist(tree_wd);
 	void const *p = rbreadlist(all);
 	while (p) {
